src/utils/schemaUtils.js

```diff
@@ -25,8 +25,53 @@ function getAllResponseContentTypes(responses) {
     return Array.from(resultSet);
 }
 
+/**
+ * recursively omit prop with the given name and if it equals to the given value
+ *
+ * @param {object} dereferencedSchema dereferenced schema
+ * @param {string} omitByPropName the prop name to omit
+ * @param {string} omitByValue omit if the prop value equals to this value
+ */
+function omitPropsFromSchema(dereferencedSchema, omitByPropName, omitByValue) {
+    if (dereferencedSchema.type === 'object') {
+        const newSchema = Object.assign({}, dereferencedSchema);
+        const schemaProperties = dereferencedSchema.properties;
+        const newSchemaProperties = Object.assign({}, schemaProperties);
+        for (const propName of Object.keys(newSchemaProperties)) {
+            if (newSchemaProperties[propName][omitByPropName] === omitByValue) {
+                // delete the prop from properties object so it would be accepted in case of additionalProperties: true
+                delete newSchemaProperties[propName];
+
+                // delete the prop from the required props
+                const index = newSchema.required ? newSchema.required.indexOf(propName) : -1;
+                if (index >= 0) {
+                    newSchema.required = newSchema.required.slice(0, index)
+                        .concat(newSchema.required.slice(index + 1));
+                }
+            } else if (newSchemaProperties[propName].type === 'object') {
+                // if the current prop is an object we need to recursively look for omitByPropName occurrences
+                newSchemaProperties[propName] = omitPropsFromSchema(newSchemaProperties[propName], omitByPropName, omitByValue);
+            }
+        }
+        newSchema.properties = newSchemaProperties;
+        return newSchema;
+    } else if (dereferencedSchema.type === 'array' && dereferencedSchema.items.type === 'object') {
+        const newSchema = Object.assign({}, dereferencedSchema);
```

Comment:
We don't need deep-cloning here?

Reply:
- I change dereferencedSchema in lines 47, 55 and 59, so I shallow clone it
- I change dereferencedSchema.properties in lines 42 and 52, so I shallow clone it
- I change newSchema.required but i'm using slice, so a new array is being created
- I don't change anything else, so we can use the original references for the rest of the objects

Something I've missed?